### ISSUE TITLE
Defer report-path allocations in Moq1001/1002, Moq1202/1204, and Moq1400/1410 hot paths

### ISSUE THREAD

**Issue by rjmurillo** [rjmurillo/moq.analyzers#1260]
## Summary

An internal code audit found three groups of analyzers that perform string formatting, LINQ chains, and redundant semantic binding **before** knowing whether any diagnostic will be reported. Since the no-diagnostic case is by far the most common (correct code), these allocations run on effectively every analyzed node:

1. **Moq1001/Moq1002** — `ConstructorArgumentsShouldMatchAnalyzer.VerifyClassMockAttempt` (`src/Analyzers/ConstructorArgumentsShouldMatchAnalyzer.cs:463-469`) builds a LINQ chain + `ToArray()` of constructors and eagerly formats the full argument string (`arguments.FormatArguments()` — a `StringBuilder` plus `Expression.ToString()` per argument) for every `new Mock<SomeClass>(...)`, undercutting the zero-allocation `FilteredArgumentList` design (`src/Common/FilteredArgumentList.cs:5-9` documents that design goal).
2. **Moq1202/Moq1204** — the Raise/Raises analyzers (`src/Analyzers/RaiseEventArgumentsShouldMatchEventSignatureAnalyzer.cs:77`, `src/Analyzers/RaisesEventArgumentsShouldMatchEventSignatureAnalyzer.cs:79`) call `EventSyntaxExtensions.GetEventNameFromSelector` before validation determines a mismatch. That helper **re-binds** the event symbol that the immediately preceding `TryGetEventMethodArgumentsFromLambdaSelector` call already resolved (a second `SemanticModel.GetSymbolInfo` via `src/Common/SemanticModelExtensions.cs:229`) and eagerly formats `eventSymbol.ToDisplayString()` (`src/Common/SemanticModelExtensions.cs:110`, `src/Common/EventSyntaxExtensions.cs:162-175`).
3. **Moq1400/Moq1410** — `SetExplicitMockBehaviorAnalyzer` (`:34`, `:45`, `:57`) and `SetStrictMockBehaviorAnalyzer` (`:58`, `:70`, `:83`) eagerly compute `GetMockedTypeName(...)` (`ToDisplayString()`) for every `new Mock<T>(...)`/`Mock.Of<T>()`/`new MockRepository(...)` in the compilation, and walk `DescendantsAndSelf().OfType<IFieldReferenceOperation>().Any(closure)` (enumerator + delegate + closure allocations) on the argument tree.

The fix is mechanical: resolve symbols once and pass them through; format strings only on the path that actually calls `ReportDiagnostic`. **All user-visible diagnostic messages stay byte-for-byte identical.**

Note (correction to a hint in the audit workflow): the Moq1202/Moq1204 message formats are **not** in `src/Analyzers/Resources.resx`; they are inline `LocalizableString` fields in the analyzers (`"Raise event arguments should match the '{0}' event delegate signature"` at `RaiseEventArgumentsShouldMatchEventSignatureAnalyzer.cs:23`, and the `Raises` variant at `RaisesEventArgumentsShouldMatchEventSignatureAnalyzer.cs:27`). `{0}` is currently filled with `eventSymbol.ToDisplayString()` — keep exactly that, just computed lazily.

### 1. `src/Analyzers/ConstructorArgumentsShouldMatchAnalyzer.cs:457-488`

```csharp
    private static void VerifyClassMockAttempt(
        SyntaxNodeAnalysisContext context,
        ITypeSymbol mockedClass,
        ArgumentListSyntax? argumentList,
        FilteredArgumentList arguments)
    {
        IMethodSymbol[] constructors = mockedClass
            .GetMembers(WellKnownMemberNames.InstanceConstructorName)
            .OfType<IMethodSymbol>()
            .Where(methodSymbol => methodSymbol.IsConstructor())
            .ToArray();

        string argumentsString = arguments.FormatArguments();
```

`argumentsString` is used only inside the two `CreateDiagnostic` calls below (`:475`, `:485`), yet it is computed for every mock creation of a class type. (`VerifyDelegateMockAttempt` at `:119` and `VerifyInterfaceMockAttempt` at `:139` already format only when a report is guaranteed — no change needed there.)

### 2. Moq1202/Moq1204 event-name double bind

`src/Analyzers/RaiseEventArgumentsShouldMatchEventSignatureAnalyzer.cs:72-79`:

```csharp
        if (!EventSyntaxExtensions.TryGetEventMethodArgumentsFromLambdaSelector(invocation, context.SemanticModel, knownSymbols, out ArgumentSyntax[] eventArguments, out ITypeSymbol[] expectedParameterTypes))
        {
            return;
        }

        string eventName = EventSyntaxExtensions.GetEventNameFromSelector(invocation, context.SemanticModel);

        context.ValidateEventArgumentTypes(eventArguments, expectedParameterTypes, invocation, Rule, eventName);
```

(Identical shape at `RaisesEventArgumentsShouldMatchEventSignatureAnalyzer.cs:74-81`.) `TryGetEventMethodArgumentsFromLambdaSelector` already resolved the `IEventSymbol` (via `TryGetEventTypeFromLambdaSelector` → private `TryGetEventSymbolFromLambdaSelector`, `SemanticModelExtensions.cs:202-237`) but only surfaces its `Type`; `GetEventNameFromSelector` then re-resolves the same symbol and formats it:

`src/Common/SemanticModelExtensions.cs:105-111`:

```csharp
    internal static bool TryGetEventNameFromLambdaSelector(
        this SemanticModel semanticModel,
        ExpressionSyntax eventSelector,
        out string? eventName)
    {
        return TryGetEventPropertyFromLambdaSelector(semanticModel, eventSelector, static eventSymbol => eventSymbol.ToDisplayString(), out eventName);
    }
```

### 3. Moq1400/Moq1410 eager type name + LINQ walks

`src/Analyzers/SetExplicitMockBehaviorAnalyzer.cs:31-61` (abridged; same pattern in `SetStrictMockBehaviorAnalyzer.cs:55-87`):

```csharp
        // Extract the type name for the diagnostic message
        string typeName = GetMockedTypeName(context.Operation, target);
        ...
        if (mockArgument?.DescendantsAndSelf().OfType<IFieldReferenceOperation>().Any(argument => argument.Member.IsInstanceOf(knownSymbols.MockBehaviorDefault)) == true)
```

The base-class helpers that finally report take the pre-formatted name (`src/Analyzers/MockBehaviorDiagnosticAnalyzerBase.cs:95-116` and `:151-162`). The base class also carries 5-parameter overloads of `TryReportMockBehaviorDiagnostic` (`:63-83`) and `TryHandleMissingMockBehaviorParameter` (`:127-138`) that have **zero callers outside each other** (grep-verified: the two derived analyzers call only the 6-parameter variants), and the 5-parameter `TryReportMockBehaviorDiagnostic` reports without message arguments even though both rules' messages contain a `{0}` placeholder — it would produce a broken message if it were ever reachable.

## Desired behavior

- No string formatting, `ToDisplayString()`, LINQ enumerator/closure allocation, or redundant `GetSymbolInfo` on the no-diagnostic path of the six analyzers above.
- Diagnostic IDs, spans, and message text are unchanged for every input.

### 1. `ConstructorArgumentsShouldMatchAnalyzer` — replace three members

```csharp
    private static void VerifyClassMockAttempt(
        SyntaxNodeAnalysisContext context,
        ITypeSymbol mockedClass,
        ArgumentListSyntax? argumentList,
        FilteredArgumentList arguments)
    {
        ImmutableArray<ISymbol> members = mockedClass.GetMembers(WellKnownMemberNames.InstanceConstructorName);
        List<IMethodSymbol> constructors = new(members.Length);
        for (int memberIndex = 0; memberIndex < members.Length; memberIndex++)
        {
            if (members[memberIndex] is IMethodSymbol methodSymbol && methodSymbol.IsConstructor())
            {
                constructors.Add(methodSymbol);
            }
        }

        // Bail out early if there are no arguments on constructors or no constructors at all
        (bool IsEmpty, Location Location) constructorIsEmpty = ConstructorIsEmpty(constructors, argumentList, context);
        if (constructorIsEmpty.IsEmpty)
        {
            // FormatArguments() allocates (StringBuilder + Expression.ToString() per argument),
            // so it is deferred to the report paths.
            Diagnostic diagnostic = constructorIsEmpty.Location.CreateDiagnostic(ClassMustHaveMatchingConstructor, mockedClass.ToDisplayString(), arguments.FormatArguments());
            context.ReportDiagnostic(diagnostic);
            return;
        }

        // We have constructors, now we need to check if the arguments match any of them
        // If the value is null it means we want to ignore and not create a diagnostic
        bool? matchingCtorFound = AnyConstructorsFound(constructors, arguments, context);
        if (matchingCtorFound.HasValue && !matchingCtorFound.Value)
        {
            Diagnostic diagnostic = constructorIsEmpty.Location.CreateDiagnostic(ClassMustHaveMatchingConstructor, mockedClass.ToDisplayString(), arguments.FormatArguments());
            context.ReportDiagnostic(diagnostic);
        }
    }

    private static (bool IsEmpty, Location Location) ConstructorIsEmpty(
        List<IMethodSymbol> constructors,
        ArgumentListSyntax? argumentList,
        SyntaxNodeAnalysisContext context)
    {
        Location location = argumentList?.GetLocation() ?? context.Node.GetLocation();
        return (constructors.Count == 0, location);
    }

    /// <summary>
    /// Checks if the provided arguments match any of the constructors of the mocked class.
    /// </summary>
    /// <param name="constructors">The constructors.</param>
    /// <param name="arguments">The arguments.</param>
    /// <param name="context">The context.</param>
    /// <returns>
    /// <see langword="true" /> if a suitable constructor was found; otherwise <see langword="false" />.
    /// If the construction method is a parenthesized lambda expression, <see langword="null" /> is returned.
    /// </returns>
    /// <remarks>Handles <see langword="params" /> and optional parameters.</remarks>
    private static bool? AnyConstructorsFound(
        List<IMethodSymbol> constructors,
        FilteredArgumentList arguments,
        SyntaxNodeAnalysisContext context)
    {
        for (int constructorIndex = 0; constructorIndex < constructors.Count; constructorIndex++)
        {
            if (IsConstructorMatch(constructors[constructorIndex], arguments, context))
            {
                return true;
            }
        }

        // Special case: parenthesized lambda creates the instance directly.
        // The compiler validates the lambda, so no additional checks are needed.
        // See https://github.com/devlooped/moq/blob/18dc7410ad4f993ce0edd809c5dfcaa3199f13ff/src/Moq/Mock%601.cs#L200
        if (arguments.Count == 1 && arguments[0].Expression.IsKind(SyntaxKind.ParenthesizedLambdaExpression))
        {
            return null;
        }

        return false;
    }
```

`IsConstructorMatch` and everything below it are unchanged. (`mockedClass.ToDisplayString()` was already report-path only.)

### 2. Moq1202/Moq1204 — resolve the `IEventSymbol` once, format on report

**`src/Common/SemanticModelExtensions.cs`** — make the existing private helper reusable. Change the declaration at `:202-206` from:

```csharp
    private static bool TryGetEventSymbolFromLambdaSelector(
        SemanticModel semanticModel,
        ExpressionSyntax eventSelector,
        out IEventSymbol? eventSymbol)
```

to:

```csharp
    internal static bool TryGetEventSymbolFromLambdaSelector(
        this SemanticModel semanticModel,
        ExpressionSyntax eventSelector,
        out IEventSymbol? eventSymbol)
```

(body unchanged; update the one internal call site in `TryGetEventPropertyFromLambdaSelector` at `:145` to extension syntax if desired). Keep `TryGetEventNameFromLambdaSelector` — it has direct unit tests (`tests/Moq.Analyzers.Test/Common/SemanticModelExtensionsTests.cs:191-260`); it simply leaves the production hot path.

**`src/Common/EventSyntaxExtensions.cs`** — replace `TryGetEventMethodArgumentsFromLambdaSelector` (`:136-154`) with a direct implementation that surfaces the symbol (the generic extractor-based `TryGetEventMethodArguments` at `:87-124` stays untouched — it has direct unit tests at `tests/Moq.Analyzers.Test/Common/EventSyntaxExtensionsTests.cs:244-405`):

```csharp
    /// <summary>
    /// Extracts event arguments from an event method invocation using the standard
    /// lambda-based event type extraction pattern shared by Raise and Raises analyzers.
    /// Resolves the event symbol exactly once so callers can defer name formatting
    /// to the diagnostic report path.
    /// </summary>
    /// <param name="invocation">The method invocation.</param>
    /// <param name="semanticModel">The semantic model.</param>
    /// <param name="knownSymbols">Known symbols for type checking.</param>
    /// <param name="eventArguments">The extracted event arguments.</param>
    /// <param name="expectedParameterTypes">The expected parameter types.</param>
    /// <param name="eventSymbol">The resolved event symbol; non-null when the method returns <see langword="true"/>.</param>
    /// <returns><see langword="true" /> if arguments were successfully extracted; otherwise, <see langword="false" />.</returns>
    internal static bool TryGetEventMethodArgumentsFromLambdaSelector(
        InvocationExpressionSyntax invocation,
        SemanticModel semanticModel,
        KnownSymbols knownSymbols,
        out ArgumentSyntax[] eventArguments,
        out ITypeSymbol[] expectedParameterTypes,
        out IEventSymbol? eventSymbol)
    {
        eventArguments = [];
        expectedParameterTypes = [];
        eventSymbol = null;

        SeparatedSyntaxList<ArgumentSyntax> arguments = invocation.ArgumentList.Arguments;

        if (arguments.Count < 1)
        {
            return false;
        }

        ExpressionSyntax eventSelector = arguments[0].Expression;
        if (!semanticModel.TryGetEventSymbolFromLambdaSelector(eventSelector, out eventSymbol) || eventSymbol is null)
        {
            return false;
        }

        expectedParameterTypes = GetEventParameterTypes(eventSymbol.Type, knownSymbols);

        if (arguments.Count > 1)
        {
            eventArguments = new ArgumentSyntax[arguments.Count - 1];
            for (int i = 1; i < arguments.Count; i++)
            {
                eventArguments[i - 1] = arguments[i];
            }
        }

        return true;
    }
```

Replace `ValidateEventArgumentTypes` (`:17-45`) so the name is formatted only when a diagnostic is created:

```csharp
    /// <summary>
    /// Validates that event arguments match the expected parameter types.
    /// </summary>
    /// <param name="context">The analysis context.</param>
    /// <param name="eventArguments">The event arguments to validate.</param>
    /// <param name="expectedParameterTypes">The expected parameter types.</param>
    /// <param name="invocation">The invocation expression for error reporting.</param>
    /// <param name="rule">The diagnostic rule to report.</param>
    /// <param name="eventSymbol">The event symbol; its display string is computed only when a diagnostic is reported.</param>
    internal static void ValidateEventArgumentTypes(
        this SyntaxNodeAnalysisContext context,
        ArgumentSyntax[] eventArguments,
        ITypeSymbol[] expectedParameterTypes,
        InvocationExpressionSyntax invocation,
        DiagnosticDescriptor rule,
        IEventSymbol? eventSymbol)
    {
        if (eventArguments.Length != expectedParameterTypes.Length)
        {
            Location location = eventArguments.Length < expectedParameterTypes.Length
                ? invocation.GetLocation()
                : eventArguments[expectedParameterTypes.Length].GetLocation();

            context.ReportDiagnostic(CreateEventDiagnostic(location, rule, GetEventDisplayName(eventSymbol)));
            return;
        }

        for (int i = 0; i < eventArguments.Length; i++)
        {
            TypeInfo argumentTypeInfo = context.SemanticModel.GetTypeInfo(eventArguments[i].Expression, context.CancellationToken);
            ITypeSymbol? argumentType = argumentTypeInfo.Type;

            if (argumentType != null && !context.SemanticModel.HasConversion(argumentType, expectedParameterTypes[i]))
            {
                context.ReportDiagnostic(CreateEventDiagnostic(eventArguments[i].GetLocation(), rule, GetEventDisplayName(eventSymbol)));
            }
        }
    }

    private static string GetEventDisplayName(IEventSymbol? eventSymbol)
    {
        // ToDisplayString() matches the string previously produced by GetEventNameFromSelector;
        // "event" matches its previous fallback, preserving message text exactly.
        return eventSymbol?.ToDisplayString() ?? "event";
    }
```

Delete `GetEventNameFromSelector` (`:162-175`) — after this change it has zero callers (grep-verified: only the two analyzers call it today, and no tests reference it). `CreateEventDiagnostic` keeps its current signature (it has direct unit tests at `EventSyntaxExtensionsTests.cs:409-460`).

**Both analyzers** — replace `Analyze` (identical shape; shown for `RaiseEventArgumentsShouldMatchEventSignatureAnalyzer.cs:62-80`, mirror for `RaisesEventArgumentsShouldMatchEventSignatureAnalyzer.cs:65-82` keeping its `IsRaisesInvocation` guard):

```csharp
    private static void Analyze(SyntaxNodeAnalysisContext context, MoqKnownSymbols knownSymbols)
    {
        InvocationExpressionSyntax invocation = (InvocationExpressionSyntax)context.Node;

        // Check if this is a Raise method call on a Mock<T>
        if (!IsRaiseMethodCall(context.SemanticModel, invocation, knownSymbols))
        {
            return;
        }

        if (!EventSyntaxExtensions.TryGetEventMethodArgumentsFromLambdaSelector(invocation, context.SemanticModel, knownSymbols, out ArgumentSyntax[] eventArguments, out ITypeSymbol[] expectedParameterTypes, out IEventSymbol? eventSymbol))
        {
            return;
        }

        context.ValidateEventArgumentTypes(eventArguments, expectedParameterTypes, invocation, Rule, eventSymbol);
    }
```

### 3. Moq1400/Moq1410 — defer `GetMockedTypeName`, remove LINQ walks

**`src/Analyzers/MockBehaviorDiagnosticAnalyzerBase.cs`** — delete the dead 5-parameter overloads (`TryReportMockBehaviorDiagnostic` `:63-83`, `TryHandleMissingMockBehaviorParameter` `:127-138`; removal is also a correctness cleanup since the 5-parameter report omits the `{0}` message argument) and replace the 6-parameter overloads so the name is computed at report time. The virtual `GetMockedTypeName` dispatch is preserved, so `SetStrictMockBehaviorAnalyzer`'s override (`:34-51`) still produces its variant names:

```csharp
    /// <summary>
    /// Attempts to report a diagnostic for a MockBehavior parameter issue.
    /// </summary>
    /// <param name="context">The operation analysis context.</param>
    /// <param name="method">The method to check for MockBehavior parameter.</param>
    /// <param name="knownSymbols">The known Moq symbols.</param>
    /// <param name="rule">The diagnostic rule to report.</param>
    /// <param name="editType">The type of edit for the code fix.</param>
    /// <param name="mockedTypeNameSource">The method whose type arguments name the mocked type; the display string is computed only when a diagnostic is actually reported.</param>
    /// <returns>True if a diagnostic was reported; otherwise, false.</returns>
    internal bool TryReportMockBehaviorDiagnostic(
        OperationAnalysisContext context,
        IMethodSymbol method,
        MoqKnownSymbols knownSymbols,
        DiagnosticDescriptor rule,
        DiagnosticEditProperties.EditType editType,
        IMethodSymbol mockedTypeNameSource)
    {
        if (!method.TryGetParameterOfType(knownSymbols.MockBehavior!, out IParameterSymbol? parameterMatch, cancellationToken: context.CancellationToken))
        {
            return false;
        }

        ImmutableDictionary<string, string?> properties = new DiagnosticEditProperties
        {
            TypeOfEdit = editType,
            EditPosition = parameterMatch.Ordinal,
        }.ToImmutableDictionary();

        string mockedTypeName = GetMockedTypeName(context.Operation, mockedTypeNameSource);
        context.ReportDiagnostic(context.Operation.CreateDiagnostic(rule, properties, mockedTypeName));
        return true;
    }

    /// <summary>
    /// Attempts to handle missing MockBehavior parameter by checking for overloads that accept it.
    /// </summary>
    /// <param name="context">The operation analysis context.</param>
    /// <param name="mockParameter">The MockBehavior parameter (should be null to trigger overload check).</param>
    /// <param name="target">The target method to check for overloads.</param>
    /// <param name="knownSymbols">The known Moq symbols.</param>
    /// <param name="rule">The diagnostic rule to report.</param>
    /// <returns>True if a diagnostic was reported; otherwise, false.</returns>
    internal bool TryHandleMissingMockBehaviorParameter(
        OperationAnalysisContext context,
        IParameterSymbol? mockParameter,
        IMethodSymbol target,
        MoqKnownSymbols knownSymbols,
        DiagnosticDescriptor rule)
    {
        // If the target method doesn't have a MockBehavior parameter, check if there's an overload that does
        return mockParameter is null
            && target.TryGetOverloadWithParameterOfType(knownSymbols.MockBehavior!, out IMethodSymbol? methodMatch, out _, cancellationToken: context.CancellationToken)
            && TryReportMockBehaviorDiagnostic(context, methodMatch, knownSymbols, rule, DiagnosticEditProperties.EditType.Insert, target);
    }

    /// <summary>
    /// Determines whether the argument's operation tree contains a reference to the given
    /// MockBehavior enum field, without LINQ/closure allocations on the hot path.
    /// </summary>
    /// <param name="argument">The argument operation to scan, or <see langword="null"/>.</param>
    /// <param name="behaviorField">The MockBehavior field symbol to look for.</param>
    /// <returns><see langword="true"/> if a matching field reference exists; otherwise <see langword="false"/>.</returns>
    private protected static bool ReferencesMockBehaviorField(IArgumentOperation? argument, IFieldSymbol? behaviorField)
    {
        if (argument is null || behaviorField is null)
        {
            return false;
        }

        foreach (IOperation descendant in argument.DescendantsAndSelf())
        {
            if (descendant is IFieldReferenceOperation fieldReference && fieldReference.Member.IsInstanceOf(behaviorField))
            {
                return true;
            }
        }

        return false;
    }
```

Message-equivalence argument (state in the PR): today both derived `AnalyzeCore` methods compute `GetMockedTypeName(context.Operation, target)` up front and pass that string into every report; the replacement computes `GetMockedTypeName(context.Operation, mockedTypeNameSource)` with `mockedTypeNameSource` always the same `target`, at report time — identical inputs, identical string.

**`src/Analyzers/SetExplicitMockBehaviorAnalyzer.cs`** — replace `AnalyzeCore`:

```csharp
    /// <inheritdoc />
    [SuppressMessage("Design", "MA0051:Method is too long", Justification = "Should be fixed. Ignoring for now to avoid additional churn as part of larger refactor.")]
    private protected override void AnalyzeCore(OperationAnalysisContext context, IMethodSymbol target, ImmutableArray<IArgumentOperation> arguments, MoqKnownSymbols knownSymbols)
    {
        // Check if the target method has a parameter of type MockBehavior
        IParameterSymbol? mockParameter = target.Parameters.DefaultIfNotSingle(parameter => parameter.Type.IsInstanceOf(knownSymbols.MockBehavior));

        // If the target method doesn't have a MockBehavior parameter, check if there's an overload that does
        if (TryHandleMissingMockBehaviorParameter(context, mockParameter, target, knownSymbols, Rule))
        {
            return;
        }

        IArgumentOperation? mockArgument = arguments.DefaultIfNotSingle(argument => argument.Parameter.IsInstanceOf(mockParameter));

        // Is the behavior set via a default value?
        if (mockArgument?.ArgumentKind == ArgumentKind.DefaultValue && mockArgument.Value.WalkDownConversion().ConstantValue.Value == knownSymbols.MockBehaviorDefault?.ConstantValue)
        {
            TryReportMockBehaviorDiagnostic(context, target, knownSymbols, Rule, DiagnosticEditProperties.EditType.Insert, target);
        }

        // NOTE: This logic can't handle indirection (e.g. var x = MockBehavior.Default; new Mock(x);). We can't use the constant value either,
        // as Loose and Default share the same enum value: `1`. Being more accurate I believe requires data flow analysis.
        //
        // The operation specifies a MockBehavior; is it MockBehavior.Default?
        if (ReferencesMockBehaviorField(mockArgument, knownSymbols.MockBehaviorDefault))
        {
            TryReportMockBehaviorDiagnostic(context, target, knownSymbols, Rule, DiagnosticEditProperties.EditType.Replace, target);
        }
    }
```

**`src/Analyzers/SetStrictMockBehaviorAnalyzer.cs`** — replace `AnalyzeCore` (the `GetMockedTypeName` override at `:34-51` stays as-is):

```csharp
    /// <inheritdoc />
    [SuppressMessage("Design", "MA0051:Method is too long", Justification = "Should be fixed. Ignoring for now to avoid additional churn as part of larger refactor.")]
    private protected override void AnalyzeCore(OperationAnalysisContext context, IMethodSymbol target, ImmutableArray<IArgumentOperation> arguments, MoqKnownSymbols knownSymbols)
    {
        // Check if the target method has a parameter of type MockBehavior
        IParameterSymbol? mockParameter = target.Parameters.DefaultIfNotSingle(parameter => parameter.Type.IsInstanceOf(knownSymbols.MockBehavior));

        // If the target method doesn't have a MockBehavior parameter, check if there's an overload that does
        if (TryHandleMissingMockBehaviorParameter(context, mockParameter, target, knownSymbols, Rule))
        {
            // Using a method that doesn't accept a MockBehavior parameter, however there's an overload that does
            return;
        }

        IArgumentOperation? mockArgument = arguments.DefaultIfNotSingle(argument => argument.Parameter.IsInstanceOf(mockParameter));

        // Is the behavior set via a default value?
        if (mockArgument?.ArgumentKind == ArgumentKind.DefaultValue && mockArgument.Value.WalkDownConversion().ConstantValue.Value == knownSymbols.MockBehaviorDefault?.ConstantValue
            && TryReportMockBehaviorDiagnostic(context, target, knownSymbols, Rule, DiagnosticEditProperties.EditType.Insert, target))
        {
            return;
        }

        // NOTE: This logic can't handle indirection (e.g. var x = MockBehavior.Default; new Mock(x);)
        //
        // The operation specifies a MockBehavior; is it MockBehavior.Strict?
        if (mockArgument?.Value.WalkDownConversion().ConstantValue.Value != knownSymbols.MockBehaviorStrict?.ConstantValue
            && !ReferencesMockBehaviorField(mockArgument, knownSymbols.MockBehaviorStrict))
        {
            TryReportMockBehaviorDiagnostic(context, target, knownSymbols, Rule, DiagnosticEditProperties.EditType.Replace, target);
        }
    }
```

Semantics check for the negated walk (state in the PR): the original expression `mockArgument?.DescendantsAndSelf()....Any(...) != true` evaluates to `true` when `mockArgument` is null; `!ReferencesMockBehaviorField(null, ...)` is also `true` — identical. Do **not** touch the `==`/`!=` boxed-constant comparisons on the `ConstantValue` lines; that is audit finding A-10 with its own issue.

## Tests

No new tests are strictly required: this change is intended to be observationally identical, and the existing suites (`ConstructorArgumentsShouldMatchAnalyzerTests` and its 8 sibling partials, `RaiseEventArgumentsShouldMatchEventSignatureAnalyzerTests`, `RaisesEventArgumentsShouldMatchEventSignatureAnalyzerTests`, `SetExplicitMockBehaviorAnalyzerTests`, `SetStrictMockBehaviorAnalyzerTests`, `MockBehaviorDiagnosticAnalyzerBaseTests`, plus `Common/EventSyntaxExtensionsTests` and `Common/SemanticModelExtensionsTests`) assert diagnostic IDs, exact spans, and messages via `{|MoqXXXX:...|}` markup — they are the regression suite and **must pass unchanged**. If any Common unit test compiles against a changed helper signature, update only the call-shape, never the assertion.

## Constraints

- Analyzer assemblies target **netstandard2.0**; Roslyn pinned to **Microsoft.CodeAnalysis 4.8** (ADR-003). No newer APIs; **no new package dependencies**. (`DescendantsAndSelf()` on `IOperation` is already used at `SetExplicitMockBehaviorAnalyzer.cs:57` and is available in 4.8.)
- **ADR-001** (name-based checks only as fast-path pre-filters before an authoritative symbol check): this issue adds **no** name-based checks; all detection remains symbol-based. Stated here because the sibling perf issue does add name pre-filters and the two may land together.
- **ADR-007** (prefer `RegisterOperationAction`): the MockBehavior analyzers already comply; the Raise/Raises/Constructor analyzers' registration model is unchanged here.
- Tests are mandatory per repo policy; here the mandate is satisfied by the existing exhaustive suites (explicitly justified above).
- User-visible messages must remain byte-for-byte identical (keep `ToDisplayString()`, lazily).

## Validation

1. `dotnet format` (no diff).
2. `dotnet build` — **0 warnings**.
3. `dotnet test --settings ./build/targets/tests/test.runsettings` — all tests pass with zero test-source changes to expectations.
4. Run the documented perf flow (`build/scripts/perf/README.md`; CONTRIBUTING.md "Performance Testing Guidelines", line 811) — relevant benchmarks exist: `tests/Moq.Analyzers.Benchmarks/Moq1002ClassBenchmarks.cs`, `Moq1202RaiseEventBenchmarks.cs`, `Moq1204RaisesEventArgumentsBenchmarks.cs`, `Moq1400ExplicitBehaviorBenchmarks.cs`, `Moq1410StrictBehaviorBenchmarks.cs`:

```powershell
./build/scripts/perf/PerfCore.ps1 -projects "tests/Moq.Analyzers.Benchmarks/Moq.Analyzers.Benchmarks.csproj" -diff
```

## Acceptance criteria

- [ ] `VerifyClassMockAttempt` performs no `FormatArguments()` call and no LINQ allocation unless a diagnostic is reported.
- [ ] Raise/Raises analyzers resolve the event symbol exactly once per invocation; `GetEventNameFromSelector` is deleted; `ToDisplayString()` runs only inside `ValidateEventArgumentTypes` report branches.
- [ ] Moq1400/Moq1410 compute `GetMockedTypeName` only inside `TryReportMockBehaviorDiagnostic` after the parameter match succeeds; the `DescendantsAndSelf().OfType<>().Any(closure)` walks are replaced by the shared allocation-free loop.
- [ ] Dead 5-parameter base-class overloads removed.
- [ ] All existing tests pass unchanged; build 0 warnings; perf comparison shows no regression.

## Out of scope

- Audit A-10 (reference-equality comparison of boxed enum `ConstantValue`s in these same analyzers) — separate correctness issue; do not change the `==`/`!=` comparisons here.
- Audit A-12/L11 (name pre-filters + cancellation tokens in the Raise analyzer and Common helpers) — separate issue; if it lands first, merge the `Analyze` methods keeping both changes.
- E-4/E-5 consolidation of the two MockBehavior analyzers/fixers — separate tech-debt item.
- `EventSetupHandlerShouldMatchEventTypeAnalyzer.cs:139` (`eventSymbol.ToDisplayString()` before validation) — same pattern, but its extraction helper is analyzer-local; can follow this issue's pattern later.
